go/login/server/server.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package backend implements GLOME-login server framework.
+package server
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+
+	"github.com/google/glome/go/glome"
+	"github.com/gorilla/mux"
+)
+
+const (
+	// MaxResponseSize is the maximum size in characaters of the response token
+	MaxResponseSize = 44
```

Comment:
Consider documenting where this comes from (32 bytes base64-encoded).

Reply:
Added.